cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -97,6 +100,130 @@ func recoverySystemEssentialSnaps(seedDir, recoverySystem string, essentialTypes
 // generateMountsMode* is called multiple times from initramfs until it
 // no longer generates more mount points and just returns an empty output.
 func generateMountsModeInstall(recoverySystem string) error {
+	var buf bytes.Buffer
+	if err := generateMountsCommonInstallRecover(recoverySystem, &buf); err != nil {
+		return err
+	}
+	if buf.Len() > 0 {
+		io.Copy(stdout, &buf)
+		return nil
+	}
+
+	// n+1: final step: write $(ubuntu_data)/var/lib/snapd/modeenv - this
+	//      is the tmpfs we just created above
+	modeEnv := &boot.Modeenv{
+		Mode:           "install",
+		RecoverySystem: recoverySystem,
+	}
+	if err := modeEnv.Write(filepath.Join(dirs.RunMnt, "ubuntu-data", "system-data")); err != nil {
+		return err
+	}
+	// and disable cloud-init in install mode
+	if err := sysconfig.DisableCloudInit(); err != nil {
+		return err
+	}
+
+	// n+2: done, no output, no error indicates to initramfs we are done
+	//      with mounting stuff
+	return nil
+}
+
+// copyUbuntuDataAuth copies the authenication files like
+//  - extrausers passwd,shadow etc
+//  - sshd host configuration
+//  - user .ssh dir
+// to the target directory. This is used to copy the authentication
+// data from a real uc20 ubuntu-data partition into a ephemeral one.
+func copyUbuntuDataAuth(src, dst string) error {
```

Comment:
This should also open the recover-ubuntu-data state.json and "copy" the user state accross so that we know if the device is managed or not so that console-conf DTRT

Reply:
we will need a helper for that, I see two options:

state.CopyState(srcStatePath, destSrcPath,  dateEntries []string) where entries is something like []string{"auth.users"}

or state.CopyState(srcStatePath, destSrcPath,  copyOver func(get func(key string, value interface{}) error , set func(key string, value interface{})) error)